lib/adhearsion-ivr/ivr_controller.rb

```diff
@@ -178,6 +192,20 @@ def prompts
       self.class.prompts
     end
 
+    def barge(val = nil)
+      if val.nil?
+        if @barge.nil?
+          self.class.barge
+        else
+          old = @barge
+          @barge = nil
```

Comment:
Also, if I'm reading this correctly, this method isn't idempotent.  Reading the value changes it, which feels dangerous.